Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -3565,3 +3626,262 @@ def remove_tag_from_endpoints_command(client: CoreClient, args: Dict):
     return CommandResults(
         readable_output=f'Successfully removed tag {tag} from endpoint(s) {endpoint_ids}', raw_response=raw_response
     )
+
+
+def parse_risky_users_or_hosts(user_or_host: dict[str, Any], table_title: str) -> dict[str, Any]:
+    reasons = user_or_host.get('reasons', [])
+    return {
+        table_title: user_or_host.get('id'),
+        'Score': user_or_host.get('score'),
+        'Description': reasons[0].get('description') if reasons else None,
+    }
+
+
+def parse_user_groups(group: dict[str, Any]) -> list[dict[str, Any]]:
+    return [
+        {
+            'User email': user,
+            'Group Name': group.get('group_name'),
+            'Group Description': group.get('description'),
+        }
+        for user in group.get("user_email", [])
+    ]
+
+
+def parse_role_names(role_data: dict[str, Any]) -> dict[str, Any]:
+    return {
+        "Role Name": role_data.get("pretty_name"),
+        "Description": role_data.get("description"),
+        "Permission": role_data.get("permissions", []),
+        "Users": role_data.get("users", []),
+        "Groups": role_data.get("groups", []),
+    }
+
+
+def enrich_error_message_id_group_role(e: DemistoException, type_: str | None, custom_message: str | None) -> str | None:
+    """
+    Handles an error that may occur during a process.
+
+    Args:
+        e (Exception): The error that occurred.
+        type (str | None): The type of resource associated with the error(Role id or Group), if applicable.
+        custom_message (str | None): A custom error message to be included in the raised ValueError, if desired.
+
+    Raises:
+        ValueError: If the error message indicates that the resource was not found, a more detailed error message
+            is constructed using the `find_the_cause_error` function and raised with the original error as the cause.
+    """
+    if (
+        e.res is not None
+        and e.res.status_code == 500
+        and 'was not found' in str(e)
+    ):
+        pattern = r"(id|Group|Role) \\?'([/A-Za-z 0-9]+)\\?'"
```

Comment:
what's the `\` before `A` for?

Reply:
1. is it `/` or `\` ? Please double-check.
2. Let's use `([A-Za-z 0-9]+(?:\\[A-Za-z 0-9]+)?)` instead, as it's more precise and won't catch `foo\`, `a\\` etc
(please test before merging)